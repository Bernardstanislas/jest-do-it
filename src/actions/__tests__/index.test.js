import * as actions from '..';
import {identity} from 'lodash';

test('loadUser action should dispatch a fetchUser action when needed', () => {
  const testUser = 'jean_moust';
  let testRequiredFields = ['name'];
  const dispatchMock = jest.fn(identity);

  const getEmptyUserStateMock = jest.fn().mockReturnValue({
    entities: {
      users: {}
    }
  });

  let thunk = actions.loadUser(testUser, testRequiredFields);
  let result = thunk(dispatchMock, getEmptyUserStateMock);

  expect(result).toMatchSnapshot();

  const getKnownUserStateMock = jest.fn().mockReturnValue({
    entities: {
      users: {
        jean_moust: {
          name: 'Jean'
        }
      }
    }
  });

  thunk = actions.loadUser(testUser, testRequiredFields);
  result = thunk(dispatchMock, getKnownUserStateMock);

  expect(result).toMatchSnapshot();

  testRequiredFields = ['name', 'age'];  

  thunk = actions.loadUser(testUser, testRequiredFields);
  result = thunk(dispatchMock, getKnownUserStateMock);

  expect(result).toMatchSnapshot();
}); 