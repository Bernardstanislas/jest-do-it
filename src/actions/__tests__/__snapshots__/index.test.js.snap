// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadUser action should dispatch a fetchUser action when needed 1`] = `
Object {
  "Call API": Object {
    "endpoint": "users/jean_moust",
    "schema": EntitySchema {
      "_getId": [Function],
      "_idAttribute": [Function],
      "_key": "users",
      "_mergeStrategy": [Function],
      "_processStrategy": [Function],
      "schema": Object {},
    },
    "types": Array [
      "USER_REQUEST",
      "USER_SUCCESS",
      "USER_FAILURE",
    ],
  },
}
`;

exports[`loadUser action should dispatch a fetchUser action when needed 2`] = `null`;

exports[`loadUser action should dispatch a fetchUser action when needed 3`] = `
Object {
  "Call API": Object {
    "endpoint": "users/jean_moust",
    "schema": EntitySchema {
      "_getId": [Function],
      "_idAttribute": [Function],
      "_key": "users",
      "_mergeStrategy": [Function],
      "_processStrategy": [Function],
      "schema": Object {},
    },
    "types": Array [
      "USER_REQUEST",
      "USER_SUCCESS",
      "USER_FAILURE",
    ],
  },
}
`;
